File: randopedia-core/src/main/java/no/extreme/randopedia/utils/DataWasher.java
package no.extreme.randopedia.utils;

import no.extreme.randopedia.model.area.Area;
import no.extreme.randopedia.model.tour.Tour;
import no.extreme.randopedia.model.tour.TourComment;

public class DataWasher {

    public Tour washTour(Tour tour) {        
        tour.setName(removeWhitespace(tour.getName()));
        tour.setName(capitalizeFirstLetter(tour.getName()));
        
        tour.setItinerary(removeWhitespace(tour.getItinerary()));
        tour.setItinerary(replaceHtmlChars(tour.getItinerary()));
        
        tour.setAccessPoint(removeWhitespace(tour.getAccessPoint()));
        tour.setAccessPoint(replaceHtmlChars(tour.getAccessPoint()));
        
        tour.setHazardsDescription(removeWhitespace(tour.getHazardsDescription()));
        tour.setHazardsDescription(replaceHtmlChars(tour.getHazardsDescription()));
        
        tour.setToolsDescription(removeWhitespace(tour.getToolsDescription()));
        tour.setToolsDescription(replaceHtmlChars(tour.getToolsDescription()));
        
        return tour;
    }
    
    public TourComment washComment(TourComment comment) {
        comment.setComment(removeWhitespace(comment.getComment()));
        return comment;
    }
    
    public Area washArea(Area area) {
        area.setName(removeWhitespace(area.getName()));
        area.setName(capitalizeFirstLetter(area.getName()));
        
        area.setDescription(removeWhitespace(area.getDescription()));
        area.setDescription(replaceHtmlChars(area.getDescription()));
        
        return area;
    }
    
    private String removeWhitespace(String text) {
        // TODO: Remove multiple white spaces in string, not just trim
        
        if(text == null) {
            return null;
        }
        
        text = text.trim();
        return text;
    }
    
    private String capitalizeFirstLetter(String text) {
        if(text == null || text.length() == 0) {
            return text;
        }
        
        text = text.substring(0, 1).toUpperCase() + text.substring(1);
        
        return text;
    }
    
    private String replaceHtmlChars(String text) {
        if(text == null || text.length() == 0) {
            return text;
        }
        
        text = text.replaceAll("<", "&lt;");
        text = text.replaceAll(">", "&gt;");
        return text;
    }
}

File: randopedia-server/helpers/data-washer.js
var dataWasher = (function() {

    function removeWhitespace(text) {
        // todo: Remove multiple white spaces, not just trim
    	return text ? text.trim() : text;
    }
    
    function capitalizeFirstLetter(text) {
        if(!text || text.length === 0) {
            return text;
        }
        return text.substring(0, 1).toUpperCase() + text.substring(1);
    }
    
    function replaceHtmlChars(text) {
        if(!text || text.length === 0) {
            return text;
        }
        
        text = text.replace("<", "&lt;");
        text = text.replace(">", "&gt;");
        return text;
    }
	
	function normalWash(text) {
        if(!text || !text.length) {
            return "";
        }

        text = removeWhitespace(text);
        text = replaceHtmlChars(text);
		text = capitalizeFirstLetter(text);

		return text;
	}
	
    function washTour(tour) {
		tour.name = normalWash(tour.name);

        tour.itineraryEng = normalWash(tour.itineraryEng);
        tour.itineraryNo = normalWash(tour.itineraryNo);

        tour.accessPointEng = normalWash(tour.accessPointEng);
        tour.accessPointNo = normalWash(tour.accessPointNo);
        
        tour.hazardsDescriptionEng = normalWash(tour.hazardsDescriptionEng);
        tour.hazardsDescriptionNo = normalWash(tour.hazardsDescriptionNo);
        
        tour.toolsDescriptionEng = normalWash(tour.toolsDescriptionEng);
        tour.toolsDescriptionNo = normalWash(tour.toolsDescriptionNo);
        
        return tour;
    }

    return {
        washTour : washTour
    };
    
})();

module.exports = dataWasher;